packages/test/test-version-utils/README.md

```diff
@@ -1,26 +1,137 @@
+
+### Mocha test setup with layer version combinations
+
+There are three compat `describe*` to generate different combinations, depending of the need of the tests
+
+`describeFullCompat`: generate test variants with compat combinations that varies the version for all layers.
+
+- Used for tests that exercise all layers and will benefits compat combinations of all layers.
+
+`describeLoaderCompat`: generate test variants with compat combinations that only varies the loader version.
+
+- Use for tests that targets the loader layer, and don't care about compat combinations of other layers.
+- Test combination generated: [CompatKind.None, CompatKind.Loader]
+
+`describeNoCompat` - generate one test variant that doesn't varies version of any layers.
+
+- Use for tests that doesn't benefit or require any compat testing.
+- Test combination generated: [CompatKind.None]
+
+These compat `describe*` functions will also load the APIs with appropriate version and provide the test with a
+`TestObjectProvider` object, where the test can use to access Fluid functionality.  Even compat testing not
+necessary, `TestObjectProvider` provide convenience function and tracking capability, and it allows the test
+to enable compat testing easily in the future just by changing the `describe*`.
+
+### Legacy version defaults and installation
+
+By default, N-1, N-2, and LTS (hard coded) test variants are generated.  The versions can be specified using command
+line (see below) to run the test against any two versions. This package includes a `mocha` global hook that will
+install legacy packages at the beginning of the package based on the `compatVersion` settings.
+
+## Command line options
+
+Tests using the compat `describe*` will be controllable using the command line options when running mocha on
+driver selection, versions for compat testing, and compat kind combinations.
+
+```text
+--compatKind <CompatKind> - filter to the compat variant. See above table. Can specify multiple times.
+                                Default: undefined (no filter)
+--compatVersion <version> - specify the old version. Relative (to base) or specific version. Can specify multiple times.
+                                Default: -1, -2, LTS (hard coded in src/compatConfig.ts)
+--baseVersion <version>   - specify the base (new) version. Allow the test to run against any version combinations.
+                                Default: <current> (same version of the test package)
+--driver <driverType>     - <driverType> = "tinylicious" | "t9s" | "routerlicious" | "r11s" | "odsp" | "local"
+                                Default: "local"
+--r11sEndpointName <name> - Determine the environment variable name to look for r11s service information to run against
+                                Default: "r11s"
+--tenantIndex <number>    - The index into the tenant list to look for odsp server information to run against
```

Comment:
Talk about how many there are?  Does it loop modulo that number or do you need to stay within those bounds?

Reply:
Yep. Modulo. Added verbiage